### ISSUE TITLE
Http/https server and domain for everyone!

### ISSUE THREAD

**Issue by mayli** [hashbang/shell-server#105]
So, let's improve the nginx on shell server to next level!

So the plan is: 

TBD

**Comment by benharri** [comment]
[admin-tools#154](https://github.com/hashbang/admin-tools/pull/154) will be helpful for this as well as [nginx/main.yml](https://github.com/hashbang/shell-server/blob/master/ansible/tasks/nginx/main.yml)

**Comment by mayli** [comment]
> [admin-tools#154](https://github.com/hashbang/admin-tools/pull/154) will be helpful for this as well as [nginx/main.yml](https://github.com/hashbang/shell-server/blob/master/ansible/tasks/nginx/main.yml)

Yep, I contributed to the original nginx service, but it turns to be not very useful to most users. This issue is to track the improvements.

**Comment by RyanSquared** [comment]
This seems like a decent roadmap but I do think we should stick with `Public` instead of `public_html` (as it is currently); as-is, people don't serve just HTML.

We already do have a systemd unit that users can enable, `SimpleHTTPServer@.service`, but that only runs HTTP and on ports >1024. Ideally, onboarding for this would be transparent as I think we already advertise that it should just work out of the box.

**Comment by benharri** [comment]
> but it turns to be not very useful to most users

dns records for *.<shellserver>.hashbang.sh haven't been deployed yet which is the main blocker

I opened a [PR for shellserver wildcard records](https://github.com/hashbang/admin-tools/pull/154)

**Comment by Arnie97** [comment]
The `SimpleHTTPServer@.service`does not seem to work now.

```
arnie97@sf1:~/Public$ systemctl start SimpleHTTPServer@8080
==== AUTHENTICATING FOR org.freedesktop.systemd1.manage-units ===
Authentication is required to start 'SimpleHTTPServer@8080.service'.
Authenticating as: root
polkit-agent-helper-1: pam_authenticate failed: Authentication failure
==== AUTHENTICATION FAILED ===
Failed to start SimpleHTTPServer@8080.service: Access denied
See system logs and 'systemctl status SimpleHTTPServer@8080.service' for details.

arnie97@sf1:~/Public$ journalctl -q
polkit-agent-helper-1[25990]: pam_unix(polkit-1:auth): auth could not identify password for [root]
```

**Comment by MineRobber9000** [comment]
@Arnie97 try:

```
systemctl --user start SimpleHTTPServer@8080
```

**Comment by RyanSquared** [comment]
Also, it's likely that port 8080 is taken. All users as of right now (and probably in the future as well) will be sharing the same address space, so it would be wise to pick one that likely isn't already used.

**Comment by RyanSquared** [comment]
As of right now, HTTPS is working. I do not see a valid solution for other peoples' domains at this point other than having some kind of domain-based challenge system.

**Comment by retronav** [comment]
Did anything progress regarding custom domains? 🙂

**Comment by RyanSquared** [comment]
We currently do not have any system set up for custom domains. At this point, I believe it would require some advanced proxy logic (dynamically looking up CNAMEs for incoming records to see which user they should go to) that's more complicated than the current setup and I don't have the time to write it.
